### DISCUSSION TITLE
Disable actions on fork by default

### DISCUSSION THREAD

**Discussion by schveiguy** [community/community#102071] [Actions]
### Select Topic Area

Product Feedback

### Body

I don't know if there's a better place to ask for this, or if it's already been asked.

But every time I fork a repository to create a PR, I get an email at some point that says my actions failed. Why? Because it's trying to run the github actions on *my fork* because there is a `.github/workflows` directory in the repository I forked. So then I go into the settings and disable actions. Fine.

But imagine all the times that someone fork and there is no error, so they get no messages. How much CPU cycles is github wasting because they are running actions on all forks? I would expect this to be a very uncommon desire (to fork and run all the actions).

Wouldn't it be more appropriate to default to actions being off on forks, and then you turn them on if you want them?

A related discussion, but was focused more on how to prevent forks from running actions: https://github.com/orgs/community/discussions/26409

What I'm asking for is for the github *project settings* to disable actions on forks by default. I think that's the more useful default, and I'd bet 99% of the time, forks would not turn them on, yet they have the option if they want to.

**Comment by ghost** [comment]
I agree that an option to globally disable actions/workflows by default for all forks should be present in the account settings. It is a waste of time/effort to disable this on every forked repo one by one. Please provide this option.

**Comment by danielbayley** [comment]
This is low hanging fruit, and would probably save so much wasted compute. Nothing compared to the vast waste on LLMs… But still an easy win.

**Comment by schveiguy** [comment]
Not sure but I think this is fixed now. I know that I’ve had to approve actions for a forked repository but I’m not sure if that was the default setting from the original.

**Comment by matinzd** [comment]
Thanks for closing this Github instead of fixing it!

**Comment by dominicegginton** [comment]
+1 for all repositories to have actions disabled for repo forks by default. 

I suggest a setting for both personal and org level accounts to be able to control with newly created repos under the accounts scope.  

I also suggest an information confirmation alert to inform a user when creating a new repo with actions enabled on forks.
